_posts/2019-10-29-dhis2-and-docker.md

```diff
@@ -0,0 +1,214 @@
+*Dockerfile* - text file, with lines and instructions on how to build the docker image. 
+
+### Docker Hub
+
+Docker Hub is an official cloud-based repository where docker users store and distribute their images. DHIS2 has its organization on [Docker Hub](https://hub.docker.com/u/dhis2/). At the time of writing this guide, we have 4  **active** repositories: base-dev, base, core-dev, core.
+
+#### Repositories
+Sometimes we categorize repositories in _channels_ and say that we have 2 image channels: _dev_ and _stable_. 
+##### dev
+Repositories suffixed with `-dev` contains images for _latest_ DHIS2 versions. That means, that we can't fully guarantee its quality and we recommend to use it only for testing purposes. Also, for curious souls, wanting to peek into the latest DHIS2 development efforts. The `-dev` repositories are ideologically the same as `dev` demo instances on _play_. Images in this channel will be overwritten multiple times a day. 
+
+##### stable
+Repositories **not** suffixed by `dev` contains images for **stable** and **released** DHIS2 versions. These images represent the stable dhis2 versions, meaning it won't t be rebuilt in the future. 
+
+##### core
+Core repositories (core, core-dev) contains tomcat images that already have DHIS2 war baked-in.  That means, that starting this image will result in a running DHIS2 instance. 
+
+##### base
+Base repositories (base, base-dev) contains only DHIS2 war. Starting this image will not result in a running DHIS2 instance. 
+
+#### Versioning and tags
+Docker tags are meant to convey useful information about a specific version/variant of the image. In DHIS2 repositories, we combine DHIS2 and tomcat versions with image variants into a tag to be as transparent as possible about what you will find in the image.
+
+`tag = dhis2 version + image variant`
+
+##### DHIS2 versions
+*dev* channels images are versioned the same way we version branches in GitHub: 
+- `master` tag will correspond to the latest *major* DHIS2 version being developed. At the time of writing this guide, the last version we released is 2.33, which means that we are currently working on the 2.34 version and that's what the version image `dhis2/core-dev:master` will contain. 
+- `2.31` to `2.33` tags will have the war of the latest *minor* dhis2 version. So, if we released 5 patches for 2.31 version in total, we are working on the 6th patch and that will be represented by `dhis2/core-dev:2.31` tag. 
+
+Versioning of *stable* images corresponds to the versioning scheme we use for DHIS2 releases and you might be already familiar with it. Examples: `2.31.5`, `2.32.1`, `2.33.0`and etc. 
+
+
+##### Image variants
+This part tells what is the image based on and what environment DHIS2 will be running in. At the time of writing this guide, we create the following image variants: 
+
+Debian: 
+- `tomcat-9.0-jdk8-openjdk-slim`
+- `tomcat-8.5-jdk8-openjdk-slim`
+- `8.5.46-jdk8-openjdk-slim`
```

Comment:
>`8.5.46-jdk8-openjdk-slim`

Isn't this also `tomcat-8.5.46-jdk8-openjdk-slim`?

Reply:
oops 👍